File: Java/src/com/algorithms/week3/MaxAdvertisementRevenueProblem.java
package com.algorithms.week3;

import java.util.Arrays;
import java.util.Scanner;

public class MaxAdvertisementRevenueProblem {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		int[] a = new int[n];
		int[] b = new int[n];
		for (int i = 0; i < n; i++)
			a[i] = scan.nextInt();
		for (int i = 0; i < n; i++)
			b[i] = scan.nextInt();

		Arrays.sort(a);
		Arrays.sort(b);
		long ans = 0;
		for (int i = 0; i < n; i++)
			ans += (long) a[i] * b[i];
		System.out.println(ans);
		scan.close();
	}

}

File: C++/week3/MaxAdvertisementRevenueProblem.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin>>n;
    int *a = new int[n], *b = new int[n];
    for (int i = 0; i < n; i++)
        cin>>a[i];
    for (int i = 0; i < n; i++)
        cin>>b[i];

    sort(a, a +n );
    sort(b, b + n);
    long ans = 0;
    for (int i = 0; i < n; i++)
        ans += (long long) a[i] * b[i];
    cout<<ans;
    return 0;
}
